nemo/collections/llm/api.py

```diff
@@ -316,11 +410,18 @@ def distill(
         >>> llm.distill(student, teacher, data, trainer, tokenizer="model")
         PosixPath('/path/to/log_dir')
     """
-    _student_model = io.load_context(student_model_path, subpath="model")
-    _teacher_model = io.load_context(teacher_model_path, subpath="model")
+    _student_model = io.load_context(ckpt_to_context_subdir(student_model_path), subpath="model")
+    _teacher_model = io.load_context(ckpt_to_context_subdir(teacher_model_path), subpath="model")
     assert isinstance(_student_model, GPTModel), "Only models based on `llm.GPTModel` are supported currently."
     assert isinstance(_teacher_model, GPTModel), "Only models based on `llm.GPTModel` are supported currently."
 
+    is_quantized_student = set_modelopt_spec_if_exists_in_ckpt(_student_model, student_model_path)
+    set_modelopt_spec_if_exists_in_ckpt(_teacher_model, teacher_model_path)
+
+    # Need to disable gradient accumulation fusion for QAT with distillation
+    if is_quantized_student:
+        _student_model.config.gradient_accumulation_fusion = False
```

Comment:
A small fix required to allow QAT with distillation. Otherwise getting this error on backward:

```python
[rank0]:   File "/opt/megatron-lm/megatron/core/tensor_parallel/layers.py", line 510, in backward
[rank0]:     if weight.main_grad.dtype == torch.float32:
[rank0]:        ^^^^^^^^^^^^^^^^
[rank0]: AttributeError: 'Tensor' object has no attribute 'main_grad'
```